services/console-proxy/server/src/main/java/com/cloud/consoleproxy/ConsoleProxyNoVncClient.java

```diff
@@ -113,25 +113,29 @@ public void run() {
                             if (client.isVncOverWebSocketConnectionOpen()) {
                                 updateFrontEndActivityTime();
                             }
-                            connectionAlive = client.isVncOverWebSocketConnectionAlive();
+                            connectionAlive = session.isOpen();
```

Comment:
@JoaoJandre @nvazquez 
would be good to use ?
```suggestion
                            connectionAlive = client.isVncOverWebSocketConnectionAlive() && session.isOpen();
```

Reply:
When I tested, ` isVncOverWebSocketConnectionAlive()`  would always return true. So I don't see a reason to use it.

Reply:
@JoaoJandre 
did you test with vmware 7.x ?
is it possible that the websocket server (esxi host) closes the console due to some reasons (e.g. timeout, or vm is stopped) ?

Reply:
@weizhouapache 

I did a quick test with VMware 7.0 U3; when the session times out on the VMware side, it closes the websocket and the resources on the CPVM are freed. 

It would be good if others could try other use cases as well.

Reply:
thanks @GutoVeronezi 
so `session.isOpen()` is good enough to determine if the connection is still alive ?